### DISCUSSION TITLE
Renamer: `Episode 1 - title_of_the_episode_1.mkv` not recognized

### DISCUSSION THREAD

**Discussion by A-Nicoladie** [Komet/MediaElch#1440] [Q&A] [answered]
Hi,

I am currently testing MediaElch and I have some questions about its use.
I currently have a directory with loose and unrenamed content :

```
- MyTVFolder
    - TV_ShowXYZ_S02.MULTI.AAC.x265.HEVC
        - Episode 1 - title_of_the_episode_1.mkv
        - Episode 2 - title_of_the_episode_2.mkv
    - [someUrl] TV_ShowXYZ_S03.x264-FINAL
        - ShowXYZ-S03E01-title_of_the_episode_1_AndSomeOtherInformations.mkv
        - ShowXYZ-S03E02-title_of_the_episode_2_AndSomeOtherInformations.mkv
```

1. When I refresh the list of series, I have the two directories in the list, but the first one is considered empty (0 episodes)... because of the file name?

2. The episodes in the first directory are not seen by MediaElch, and cannot be renamed... how can I rename the files automatically?

3. When I ask to rename (`<showTitle> - S<season>E<episode> - <title>.<extension>`) the files in the second directory, they are renamed as follows:
`[someUrl] TV_ShowXYZ_S03.x264-FINAL - S03E01 - ShowXYZ-S03E01-title_of_the_episode_1_AndSomeOtherInformations.mkv`

4. The two directories belong to the same series, but they are not renamed (yes I have checked "rename folder" `<title> (<year>)`. How I can force this?

> MediaElch 2.8.16 (appimage)
> Ubuntu 22.04

Thanks for your help!

**Comment by bugwelle** [comment, marked as answer]
Hi there,

sorry for the late answer. I mostly check MediaElch once a week (or two weeks). See [Office Hours](https://github.com/Komet/MediaElch/issues/1417).

To answer your questions:

1. It's because of the episode's filename. If you use `S02E01`, etc. it should work. 
2. Only files that are recognized are renamed. We don't rename "unknown" files.
3. Have you scraped TV show _and_ episode details? It looks as if the foldername is used (the default).
4. Will need to check. Please first scrape the TV show _and_ episode and try again. It may work then (though unchecked, would need to check).

Regards,
Andre

**Comment by A-Nicoladie** [comment]
Don't worry, there's nothing urgent. Priority to family life and work.

> 1. It's because of the episode's filename. If you use S02E01, etc. it should work.
> 2. Only files that are recognized are renamed. We don't rename "unknown" files.

Yes. It will be good to have a "detector" for misnamed files (as an enhancement).
It would allow to propose a corrected file name and avoid having to rename everything by hand.

e.g. in my case:  
_If the file name contains "episode xx" or "Epxx" or "Exx" or "yxx" or...  
-> then it is probably episode XX.  
Then MediaElch can check the season by searching in the file or folder name  "Saison yy" or "Syy" or "yxx
-> so it's probably season YY_

> 3. Have you scraped TV show and episode details? It looks as if the foldername is used (the default).
> 4. Will need to check. Please first scrape the TV show and episode and try again. It may work then (though unchecked, would need to check).

You are right. I forgot to save changes before trying to rename. It works fine.

Thanks for your time. :+1:

**Reply by bugwelle** [reply]
Thanks. :-)

A "unused files" detector was requested in another feature request, I think. I never got to implement it, though.
